assets/stylesheets/_error.scss

```diff
@@ -23,12 +23,12 @@
 .c-error-page__header {
     font-size: floor($font-size-base * 4);
     font-weight: 700;
-    color: $white;
+    color: $black;
 }
 
 .c-error-page__subheader {
     font-size: floor($font-size-base * 3);
-    color: $white;
+    color: $black;
 }
```

Comment:
I think you can just remove these `color` entries and it'll pick up the global text colour, which should be `$gray700` instead of `$black`